### PR TITLE
3/3 for alpha interface updates -- CD workflows

### PR DESCRIPTION
**Pull request by Ryanmtate**
This PR will build documentation and push to the `docs` branch to be served by github pages.

Additionally, this will publish the node package to npm.

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Publish
+        run: |
+           make run_publish
+  push_docs:
+    name: build and push static docs
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node: [ '12', '14' ]
```
Comment (sehz) [inline .github/workflows/cd.yml:36]: restrict to a single runtime

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    steps:
+      - uses: actions/checkout@master
```
Comment (sehz) [inline .github/workflows/cd.yml:38]: use release version of checkout

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        with:
+          persist-credentials: false # otherwise, the token used is the GITHUB_TOKEN, instead of your personal token
+          fetch-depth: 0 # otherwise, you will failed to push refs to dest repo
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Publish
+        run: |
+           make run_publish
```
Comment (sehz) [inline .github/workflows/cd.yml:29]: this should not be trigger for every push but for release. see example of how to trigger on tags: https://github.com/actions/create-release

File: .github/workflows/cd.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Build docs
+        run: |
+          make run_docs
+      - name: Commit files
+        run: |
+          git add --force docs
+          git config --local user.email "action@github.com"
```
Comment (sehz) [inline .github/workflows/cd.yml:52]: this should be fluvio authors

File: .github/workflows/cd.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          persist-credentials: false # otherwise, the token used is the GITHUB_TOKEN, instead of your personal token
+          fetch-depth: 0 # otherwise, you will failed to push refs to dest repo
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Build docs
+        run: |
+          make run_docs
+      - name: Commit files
```
Comment (sehz) [inline .github/workflows/cd.yml:49]: shouldn't be checking out doc branch?

**Review by sehz — CHANGES_REQUESTED** [review]

File: .github/workflows/cd.yml (hunk 5 above)
Comment (Ryanmtate) [reply .github/workflows/cd.yml:47]: commit 072af98 checks out docs branch

File: .github/workflows/cd.yml (hunk 4 above)
Comment (Ryanmtate) [reply .github/workflows/cd.yml]: commit 072af98 updates commit author to fluvio authors

File: .github/workflows/cd.yml (hunk 3 above)
Comment (Ryanmtate) [reply .github/workflows/cd.yml:27]: commit 072af98 now runs workflow on pushing a tag;

File: .github/workflows/cd.yml (hunk 2 above)
Comment (Ryanmtate) [reply .github/workflows/cd.yml]: commit 072af98 now uses `v2`

File: .github/workflows/cd.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+jobs:
+  publish:
+    name: Publish Fluvio Client to NPM
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node: [ '12', '14' ]
+    steps:
+      - uses: actions/checkout@master
```
[the remark this answers is not in the document]
Comment (Ryanmtate) [reply .github/workflows/cd.yml]: commit 072af98 now uses `v2`

File: .github/workflows/cd.yml (hunk 1 above)
Comment (Ryanmtate) [reply .github/workflows/cd.yml]: commit 072af98 uses single runtime

File: .github/workflows/cd.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        with:
+          persist-credentials: false # otherwise, the token used is the GITHUB_TOKEN, instead of your personal token
+          fetch-depth: 0 # otherwise, you will failed to push refs to dest repo
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
```
[the remark this answers is not in the document]
Comment (Ryanmtate) [reply .github/workflows/cd.yml:24]: commit 072af98 uses single runtime

File: .github/workflows/cd.yml [hunk 8]
```diff
@@ -0,0 +1,59 @@
+name: CD
+
+# See article for setting up github site pages for docs;
+# https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site
```
[the remark this answers is not in the document]
Comment (Ryanmtate) [reply .github/workflows/cd.yml]: commit 072af98 removes;

**Comment by sehz** [comment]
Can you test this workflow?

**Comment by Ryanmtate** [comment]
> Can you test this workflow?

Needs to be merge before I can trigger a workflow

**Comment by sehz** [comment]
Can you split workflow so they can tested independently?

And should add alpha.1 like that for npm version

**Comment by Ryanmtate** [comment]
> Can you split workflow so they can tested independently?
> 
> And should add alpha.1 like that for npm version

commit 6ecc88d splits into `docs` and `publish` workflows; adds `alpha.1` to package.json version

**Review by sehz — APPROVED** [review]

**Pull request merged**